tests/manager/integration/license/overdrive/test_importer.py

```diff
@@ -0,0 +1,985 @@
+"""Tests for the OverdriveImporter class."""
+
+from __future__ import annotations
+
+from typing import Any
+from unittest.mock import AsyncMock, Mock
+
+import pytest
+
+from palace.manager.core.exceptions import PalaceValueError
+from palace.manager.data_layer.bibliographic import BibliographicData
+from palace.manager.data_layer.circulation import CirculationData
+from palace.manager.integration.license.overdrive.api import (
+    BookInfoEndpoint,
+    OverdriveAPI,
+)
+from palace.manager.integration.license.overdrive.importer import (
+    FeedImportResult,
+    OverdriveImporter,
+)
+from palace.manager.integration.license.overdrive.representation import (
+    OverdriveRepresentationExtractor,
+)
+from palace.manager.service.redis.models.set import IdentifierSet
+from palace.manager.sqlalchemy.model.coverage import Timestamp
+from palace.manager.sqlalchemy.model.identifier import Identifier
+from palace.manager.util.datetime_helpers import datetime_utc
+from tests.fixtures.database import DatabaseTransactionFixture
+from tests.fixtures.overdrive import OverdriveAPIFixture
+from tests.fixtures.services import ServicesFixture
+
+
+def _create_mock_book_data(
+    book_id: str,
+    has_metadata: bool = True,
+    has_availability: bool = True,
+    date_added: str | None = None,
+) -> dict[str, Any]:
+    """Helper to create mock book data for tests.
+
+    Args:
+        book_id: The OverDrive book ID
+        has_metadata: Whether to include metadata in the book data
+        has_availability: Whether to include availability data
+        date_added: Optional date_added timestamp string
+
+    Returns:
+        Mock book data dictionary
+    """
+    book: dict[str, Any] = {"id": book_id}
+    if has_metadata:
+        book["metadata"] = {"title": f"Book {book_id}"}
+    else:
+        book["metadata"] = None
+    if has_availability:
+        book["availabilityV2"] = {"copiesOwned": 1, "copiesAvailable": 1}
+    if date_added:
+        book["date_added"] = date_added
+    return book
```

Comment:
Instead of mocking the book data like this, can we actually use some of the real Overdrive book data we have in the files fixture?

Reply:
Yes - I will fix that.